server/datastore/mysql/microsoft_mdm.go

```diff
@@ -617,6 +617,298 @@ func (ds *Datastore) DeleteMDMWindowsConfigProfile(ctx context.Context, profileU
+                AND NOT EXISTS (
+                    SELECT
+                        1 FROM host_mdm_windows_profiles hmwp2
+                    WHERE (h.uuid = hmwp2.host_uuid
+                        AND hmwp2.status = ?))`
+	args := []interface{}{
+		fleet.MDMDeliveryPending,
+		fleet.MDMDeliveryFailed,
+	}
+	return sql, args
+}
+
+func subqueryHostsMDMWindowsOSSettingsStatusVerifying() (string, []interface{}) {
+	sql := `
+            SELECT
+                1 FROM host_mdm_windows_profiles hmwp
+            WHERE
+                h.uuid = hmwp.host_uuid
+                AND hmwp.operation_type = ?
+                AND hmwp.status = ?
+                AND NOT EXISTS (
+                    SELECT
+                        1 FROM host_mdm_windows_profiles hmwp2
+                    WHERE (h.uuid = hmwp2.host_uuid
+                        AND hmwp2.operation_type = ?
+                        AND(hmwp2.status IS NULL
+                            OR hmwp2.status NOT IN(?, ?))))`
+
+	args := []interface{}{
+		fleet.MDMOperationTypeInstall,
+		fleet.MDMDeliveryVerifying,
+		fleet.MDMOperationTypeInstall,
+		fleet.MDMDeliveryVerifying,
+		fleet.MDMDeliveryVerified,
+	}
+	return sql, args
+}
+
+func subqueryHostsMDMWindowsOSSettingsStatusVerified() (string, []interface{}) {
+	sql := `
+            SELECT
+                1 FROM host_mdm_windows_profiles hmwp
+            WHERE
+                h.uuid = hmwp.host_uuid 
+                AND hmwp.operation_type = ?
+                AND hmwp.status = ?
+                AND NOT EXISTS (
+                    SELECT
+                        1 FROM host_mdm_windows_profiles hmwp2
+                    WHERE (h.uuid = hmwp2.host_uuid
+                        AND hmwp2.operation_type = ?
+                        AND(hmwp2.status IS NULL
+                            OR hmwp2.status != ?)))`
+	args := []interface{}{
+		fleet.MDMOperationTypeInstall,
+		fleet.MDMDeliveryVerified,
+		fleet.MDMOperationTypeInstall,
+		fleet.MDMDeliveryVerified,
+	}
+	return sql, args
+}
+
+func (ds *Datastore) GetMDMWindowsProfilesSummary(ctx context.Context, teamID *uint) (*fleet.MDMProfilesSummary, error) {
+	includeBitLocker, err := ds.getConfigEnableDiskEncryption(ctx, teamID)
+	if err != nil {
+		return nil, err
+	}
+
+	var counts []statusCounts
+	if !includeBitLocker {
+		counts, err = getMDMWindowsStatusCountsProfilesOnlyDB(ctx, ds, teamID)
+	} else {
+		counts, err = getMDMWindowsStatusCountsProfilesAndBitLockerDB(ctx, ds, teamID)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	var res fleet.MDMProfilesSummary
+	for _, c := range counts {
+		switch c.Status {
+		case "failed":
+			res.Failed = c.Count
+		case "pending":
+			res.Pending = c.Count
+		case "verifying":
+			res.Verifying = c.Count
+		case "verified":
+			res.Verified = c.Count
+		default:
+			level.Debug(ds.logger).Log("msg", "unexpected mdm windows status count", "status", c.Status, "count", c.Count)
```

Comment:
I wonder if this should not be raised more loudly, either by returning an error or logging as error?

Reply:
Unless it's not that unusual - I see that it's the `ELSE` in the `CASE` of the main select, maybe a host that doesn't have any profile assigned would result in this case?